src/components/trade/PriceImpact/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import BigNumber from 'bignumber.js'
+import BN from 'bn.js'
+import { TokenDex } from '@gnosis.pm/dex-js'
+
+import { BoldColourTag } from './PriceImpact.styled'
+import { FormMessage } from 'components/common/FormMessage'
+import { HelpTooltip, HelpTooltipContainer } from 'components/Tooltip'
+import { PriceSuggestionsWrapper } from '../PriceSuggestions/PriceSuggestions'
+
+import useBestAsk from 'hooks/useBestAsk'
+
+import { formatSmart, parseAmount } from 'utils'
+import { calculatePriceImpact, determinePriceWarning, getImpactColourClass } from './utils'
+
+export interface PriceImpactArgsBase {
+  limitPrice?: string | null
+  bestAskPrice: BigNumber | null
+}
+
+export interface PriceImpactArgs extends PriceImpactArgsBase {
+  fillPrice: BigNumber | null
+}
+
+interface PriceImpactProps {
+  baseToken: TokenDex
+  quoteToken: TokenDex
+  limitPrice?: string | null
+  fillPrice: BigNumber | null
+  networkId: number
+}
+
+const PriceImpactTooltip: React.FC = () => (
+  <HelpTooltipContainer>
+    The difference between the market price and the limit price due to order size
+  </HelpTooltipContainer>
+)
+
+function PriceImpact({
+  limitPrice,
+  fillPrice,
+  baseToken,
+  quoteToken,
+  networkId,
+}: PriceImpactProps): React.ReactElement | null {
+  const { id: baseTokenId } = baseToken
+  const { id: quoteTokenId } = quoteToken
+
+  // TODO: useBestAskPrice hook here
+  const { bestAskPrice } = useBestAsk({
+    networkId,
+    baseTokenId,
+    quoteTokenId,
+  })
+
+  const { priceImpactSmart, className, priceWarning } = React.useMemo(() => {
+    const priceImpact = calculatePriceImpact({ bestAskPrice, limitPrice })
+    // Smart format, if possible
+    const priceImpactSmart =
+      (priceImpact &&
+        !priceImpact.isZero() &&
+        formatSmart({ amount: parseAmount(priceImpact.toString(), 4) as BN, precision: 4, smallLimit: '0.01' })) ||
+      '<0.01'
```

Comment:
I personally don't love this long complex lines. I feel that, `AA && BB || CC` pattern are meant for simpler expressions. Also, good to add line breaks when you make calls and u have IF/ELSE logics

One suggestion is:
```suggestion
    let priceImpactSmart
    if (priceImpact && !priceImpact.isZero()) {
      priceImpactSmart = formatSmart({
        amount: parseAmount(priceImpact.toString(), 4) as BN,
        precision: 4,
        smallLimit: '0.01',
      })
    } else {
      priceImpactSmart = '<0.01'
    }
```

Or even, it's possible to take my suggestion and move it (with the calculatePriceImpact too) to it's own function, so you also simplify the hook, maybe even the whole hook content

Reply:
I kinda want to adopt [`do` ](https://babeljs.io/docs/en/babel-plugin-proposal-do-expressions) expressions for cases like this
But only kinda

Reply:
made it an entire hook in another PR, good advice thanks